chromium_src/chrome/browser/component_updater/widevine_cdm_component_installer.cc

```diff
@@ -3,39 +3,339 @@
+    LOG(ERROR) << "Arm64 DLL download failed. Error: "
+               << net::ErrorToShortString(error);
+    result_ = update_client::ToInstallerResult(error);
+  } else {
+    bool success = ExtractArm64Dll(zip_path) && AddArm64ArchToManifest();
+    if (!success) {
+      result_ =
+          CrxInstaller::Result(update_client::InstallError::GENERIC_ERROR);
+    }
+  }
+  // The loader needs to be destroyed on the current sequence. We therefore
+  // need to do it here, and cannot wait until the destructor does it.
+  loader_.reset();
+  installed_.Signal();
+}
+
+bool WidevineArm64DllInstaller::ExtractArm64Dll(base::FilePath zip_path) {
+  VLOG(2) << "Extracting Arm64 DLL.";
+  base::FilePath arm64_directory = GetPlatformDirectory(install_dir_);
+  base::File::Error error;
+  if (base::CreateDirectoryAndGetError(arm64_directory, &error)) {
+    if (!zip::Unzip(zip_path, arm64_directory)) {
+      LOG(ERROR) << "Failed to unzip Arm64 DLL.";
+      return false;
+    }
+  } else {
+    LOG(ERROR) << "Failed to create " << arm64_directory << ": "
+               << base::File::ErrorToString(error);
+    return false;
+  }
+  return true;
+}
+
+// Components contain a "manifest.json" file that has a list "accept_arch",
+// which lists the supported architectures. The file is checked on browser
+// startup when the components are registered. If the browser's architecture is
+// not in the list of supported architectures, then the component is
+// uninstalled. To prevent this from happening to our WIDEVINE_ARM64_DLL_FIX,
+// we need to add "arm64" to the list of supported architectures.
+bool WidevineArm64DllInstaller::AddArm64ArchToManifest() {
+  VLOG(2) << "Adding Arm64 to manifest.json.";
+  base::FilePath manifest_path = install_dir_.AppendASCII("manifest.json");
+
+  std::string json_content;
+  if (!base::ReadFileToString(manifest_path, &json_content)) {
+    LOG(ERROR) << "Failed to read file: " << manifest_path;
+    return false;
+  }
+
+  absl::optional<base::Value> root = base::JSONReader::Read(json_content);
+  if (!root) {
+    LOG(ERROR) << "Failed to parse JSON.";
+    return false;
+  }
+
+  base::Value::Dict* root_dict = root->GetIfDict();
+  if (!root_dict) {
+    LOG(ERROR) << "Manifest is not a dictionary.";
+    return false;
+  }
+
+  base::Value* accept_arch = root_dict->Find("accept_arch");
+  if (!accept_arch) {
+    LOG(ERROR) << "Could not find accept_arch field.";
+    return false;
+  }
+
+  base::Value::List* accept_arch_list = accept_arch->GetIfList();
+  if (!accept_arch_list) {
+    LOG(ERROR) << "accept_arch is not a list.";
+    return false;
+  }
+
+  accept_arch_list->Append("arm64");
+
+  std::string new_json_content;
+  base::JSONWriter::Write(*root, &new_json_content);
+
+  if (!base::WriteFile(manifest_path, new_json_content)) {
+    LOG(ERROR) << "Failed to write file: " << manifest_path;
+    return false;
+  }
+
+  return true;
+}
+
+void RegisterWidevineCdmComponent(
```

Comment:
why did we add a second copy of this method instead of using buildflags around the params? It's very confusing, especially because the two methods are not even together in the file. This one also has a VLOG that the other one doesn't

Reply:
The motivation was to split this source file into two clean halves: One with `BUILDFLAG(WIDEVINE_ARM64_DLL_FIX)` and the other one without. It felt easier / clearer when implementing this PR. But I can see now that for example when searching for the definition of the function, it is less clear to see which definition will get used. Do you want me to file a follow-up PR that refactors this?